spec/ruby/optional/capi/class_spec.rb

```diff
@@ -385,17 +385,6 @@
         CApiClassSpecs.const_get(cls.name)
       }.should raise_error(NameError, /wrong constant name/)
     end
-
-    ruby_version_is "3.5" do
-      it "calls .inherited before .const_added" do
-        ScratchPad.record([])
-        @s.rb_define_class_id_under(CApiClassSpecs::Callbacks, :Subclass, CApiClassSpecs::Callbacks)
-        ScratchPad.recorded.should == [
-          [:inherited, "CApiClassSpecs::Callbacks::Subclass", :location],
-          [:const_added, :Subclass, :location],
-        ]
-      end
-    end
```

Comment:
I think we should keep these specs, just update them to match the old behavior.

Reply:
I have that one in my [original/followup patch](https://github.com/ruby/ruby/pull/12759/files).

Reply:
Let me add that my test verifies exactly callback execution order, which is what the test is about. I'll add @eregon suggestion too.

Then, I could add tests that say: In both callbacks, inheritance has happened, and constant assignment has happened (there is a location, `constants` has the constant, the class has a permanent name, etc.).